script.js

```diff
@@ -1,4 +1,282 @@
+var index = 0;
+var index2 = 0;
+var index4 = 0;
+index3 = 0;
+var mode = "first deal";
+var cardDeck = [];
+var playerRank = 0;
+var computerRank = 0;
+var initialMode = `first deal`;
+var currentPlayerRank = 0;
+
+// currentPlayerRank=checkRank(playerCard,playerRankArray,playerRank)
+// currentPlayerRank=checkRank(computerCard,computerRankArray,computerRank)
+// checkBust(playerRank,`player`,'computer')
+// checkBust(computerRank,`computer`,'player')
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  //make the deck
+  cardDeck = makeDeck();
+  shuffledCardDeck = shuffleCards(cardDeck);
+  if (mode == `value of ace`) {
+    if (input != `1` && input != `11`) {
+      return `Please enter either '1' or '11'`;
+    } else {
+      changingValueOfAce(input);
+    }
+  }
+  if (input == `play again`) {
+    others = ``;
+    playerCard = [];
+    computerCard = [];
+    cardDeck = makeDeck();
+    playerRank = 0;
+    computerRank = 0;
+    playerRankArray = [];
+    computerRankArray = [];
+    mode = `first deal`;
+    index2 = 0;
+    index3 = 0;
+    index = 0;
+  }
+  if (mode == "first deal") {
+    //deal one card to player and computer
+    dealACard(playerCard);
+    dealACard(computerCard);
+    playerRank += playerCard[index2].rank;
+    computerRank += computerCard[index4].rank;
+    index4 += 1;
+    if (playerCard[index2].name == `ace`) {
+      mode = `value of ace`;
+      index2 += 1;
+      return standardReply(
+        "Type '1' if you want the value of Ace to be 1 and '11' if you want it to be 11"
+      );
+    } else {
+      index2 += 1;
+      checkBust();
+      console.log(playerCard);
+      console.log(computerCard);
+      mode = "second deal";
+      return standardReply(`Press submit to get another card.`);
+    }
+  }
+  if (mode == "second deal") {
+    initialMode = `second deal`;
+    dealACard(playerCard);
+    dealACard(computerCard);
+    playerRank += playerCard[index2].rank;
+    computerRank += computerCard[index4].rank;
+    if (playerCard[index2].name == `ace`) {
+      mode = `value of ace`;
+      index2 += 1;
+      index4 += 1;
+      return standardReply(
+        "Type '1' if you want the value of Ace to be 1 and '11' if you want it to be 11"
+      );
+    } else {
+      index4 += 1;
+      index2 += 1;
+      checkBust();
+      console.log(playerCard);
+      console.log(computerCard);
+      mode = `subsequent deal`;
+      return standardReply(
+        ` Enter 'hit' to get another card or 'stay' if you dont want another card.`
+      );
+    }
+  }
+  if (mode == `subsequent deal`) {
+    if (input == ``) {
+      return standardReply(
+        ` Enter 'hit' to get another card or 'stay' if you dont want another card.`
+      );
+    }
+    if (input == "stay") {
+      if (computerRank <= 16) {
+        dealACard(computerCard);
+        computerRank += computerCard[index2].rank;
+        checkBust();
+        return `${whoWins()}`;
```

Comment:
you dont necessarily need to use the template string literals. You could just do the following:
return whoWins()